SUPPORTED_APIS.md

```diff
@@ -2,31 +2,122 @@
+| `HARDHAT` | `hardhat_setCoinbase` | NOT IMPLEMENTED | Sets the coinbase address |
+| `HARDHAT` | `hardhat_setLoggingEnabled` | NOT IMPLEMENTED | Enables or disables logging |
+| `HARDHAT` | `hardhat_setMinGasPrice` | NOT IMPLEMENTED | Sets the minimum gas price |
+| `HARDHAT` | `hardhat_setNextBlockBaseFeePerGas` | NOT IMPLEMENTED | Sets the base fee per gas for the next block |
+| `HARDHAT` | `hardhat_setPrevRandao` | NOT IMPLEMENTED | Sets the RANDAO value of the next block |
+| `HARDHAT` | `hardhat_setNonce` | NOT IMPLEMENTED | Sets the nonce of a given account |
+| `HARDHAT` | `hardhat_setStorageAt` | NOT IMPLEMENTED | Sets the storage value at a given key for a given account |
+| `HARDHAT` | `hardhat_stopImpersonatingAccount` | Stop impersonating an account after having previously used |
+| `EVM` | `evm_addAccount` | NOT IMPLEMENTED | Adds any arbitrary to the namespace |
+| `EVM` | `evm_increaseTime` | NOT IMPLEMENTED | Jump forward in time by the given amount of time |
+| `EVM` | `evm_mine` | NOT IMPLEMENTED | Force a single block to be mined |
+| `EVM` | `evm_removeAccount` | NOT IMPLEMENTED | Removes an account from the namespace |
+| `EVM` | `evm_revert` | NOT IMPLEMENTED | Revert the state of the blockchain to a previous snapshot |
+| `EVM` | `evm_setAccountBalance` | NOT IMPLEMENTED | Sets the given account's balance to the specified WEI value |
+| `EVM` | `evm_setAccountCode` | NOT IMPLEMENTED | Sets the given account's code to the specified data |
+| `EVM` | `evm_setAccountNonce` | NOT IMPLEMENTED | Sets the given account's nonce to the specified value |
+| `EVM` | `evm_setAccountStorageAt` | NOT IMPLEMENTED | Sets the given account's storage slot to the specified data |
+| `EVM` | `evm_setTime` | NOT IMPLEMENTED | Sets the internal clock time to the given timestamp |
+| `EVM` | `evm_snapshot` | NOT IMPLEMENTED | Snapshot the state of the blockchain at the current block |
+| `ETH` | `eth_accounts` | NOT IMPLEMENTED | Returns a list of addresses owned by client |
+| `ETH` | `eth_blockNumber` | NOT IMPLEMENTED | Returns the number of the most recent block |
+| `ETH` | `eth_call` | NOT IMPLEMENTED | Executes a new message call immediately without creating a transaction on the block chain |
+| `ETH` | `eth_chainId` | NOT IMPLEMENTED | Returns the currently configured chain id |
+| `ETH` | `eth_coinbase` | NOT IMPLEMENTED | Returns the client coinbase address |
+| `ETH` | `eth_estimateGas` | NOT IMPLEMENTED | Generates and returns an estimate of how much gas is necessary for the transaction to complete |
+| `ETH` | `eth_feeHistory` | NOT IMPLEMENTED | Returns a collection of historical block gas data |
+| `ETH` | `eth_gasPrice` | NOT IMPLEMENTED | Returns the current price per gas in wei |
+| `ETH` | `eth_getBalance` | NOT IMPLEMENTED | Returns the balance of the account of given address |
+| `ETH` | `eth_getBlockByHash` | NOT IMPLEMENTED | Returns information about a block by block hash |
```

Comment:
This is `IMPLEMENTED`: https://github.com/matter-labs/era-test-node/blob/main/src/node.rs#L699

Reply:
For this one I get the following error:

```bash
"error": {
    "code": -32601,
    "message": "Method get_block_by_hash__0x0000…0000 is not implemented"
  }
  ```

Reply:
Request was:

```bash
POST http://localhost:8011
content-type: application/json

{
    "jsonrpc": "2.0",
    "id": "1",
    "method": "eth_getBlockByHash",
    "params": ["0x0000000000000000000000000000000000000000000000000000000000000000", true]
}
```

Reply:
Oh interesting... perhaps the hash is incorrect for the block that you're looking for. It will only return a Block if the Hash exists in memory.

This can also be categorized `PARTIALLY` as we're not handling the response correctly if the Hash does not match a block in memory

Reply:
I lean towards leaving it at `NOT IMPLEMENTED` then

Reply:
Tracked here in this Issue: https://github.com/matter-labs/era-test-node/issues/25